exercises/sieve/example.rs

```diff
@@ -0,0 +1,26 @@
+pub struct Sieve { limit: usize }
+
+impl Sieve {
+    pub fn new(limit: usize) -> Sieve {
+        Sieve { limit: limit }
+    }
+
+    pub fn primes(&self) -> Vec<usize> {
+        let mut integers = (1..self.limit).map ( |x| x + 1).collect::<Vec<usize>>();
+        let mut p = 2;
+        let mut done = false;
+
+        while !done {
+            integers.retain( |&x| (x == p) || (x % p != 0));
+
+            let next_possible_prime = integers.iter().find( |&x| *x > p);
+
+            if next_possible_prime.is_none() {
```

Comment:
also don't feel particularly strongly, but what if we used the `let Some(next_prime) = ...`  form here? Could possibly even fold line 16 into this. Would avoid the `unwrap`, so I think looks slightly better